src/isomorphic/children/ReactChildren.js

```diff
@@ -117,7 +117,7 @@ function mapSingleChildIntoContext(bookKeeping, child, childKey) {
         // traverseAllChildren used to do for objects as children
         keyPrefix +
         (
-          mappedChild !== child ?
+          mappedChild !== child && mappedChild.key ?
           escapeUserProvidedKey(mappedChild.key || '') + '/' :
```

Comment:
Assume Ben likes this, can you drop the `|| ''` here since we already know we have it.